src/server/RedirectingHttpHandler.ts

```diff
@@ -0,0 +1,70 @@
+import { getLoggerFor } from '../logging/LogUtil';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import type { HttpHandlerInput } from './HttpHandler';
+import { HttpHandler } from './HttpHandler';
+
+export class RedirectingHttpHandler extends HttpHandler {
+  private readonly logger = getLoggerFor(this);
+  private readonly matcher: RegExp;
+  private readonly redirects: { pattern: RegExp; redirect: string }[] = [];
+
+  /**
+   * Creates a handler for the provided redirects.
+   * @param redirects - A mapping between URL patterns.
+   */
+  public constructor(redirects: Record<string, string>, private readonly statusCode: number = 308) {
+    super();
+    const patterns = Object.keys(redirects);
+
+    // Create a single regexp for quick checks
+    this.matcher = new RegExp(`^${patterns.join('|')}$`, 'u');
+
+    // Create an array of (regexp, redirect) pairs
+    this.redirects = patterns.map(
+      (pattern): { pattern: RegExp; redirect: string } => ({
+        pattern: new RegExp(pattern, 'u'),
+        redirect: redirects[pattern],
+      }),
+    );
+  }
+
+  public async canHandle({ request }: HttpHandlerInput): Promise<void> {
+    // Only return if a redirect is configured for the requested URL
+    if (!this.matcher.test(request.url ?? '')) {
+      throw new NotImplementedHttpError(`No redirect configured for ${request.url}`);
+    }
+  }
+
+  public async handle({ request, response }: HttpHandlerInput): Promise<void> {
+    const { url } = request;
+
+    // Get groups and redirect of first matching pattern
+    const result = this.redirects.reduce<{ match: RegExpExecArray; redirect: string } | null>(
+      (prev, { pattern, redirect }): { match: RegExpExecArray; redirect: string } | null => {
+        if (prev) {
+          return prev;
+        }
+        const match = pattern.exec(url ?? '');
+        if (match) {
+          return { match, redirect };
+        }
+        return null;
+      },
+      null,
+    );
```

Comment:
Might be better done with a simple loop, given the early return? (took me a while to parse)

Reply:
Yeah, wondered that myself, but then it was already written. Still prefer this reduction over using mutable variables, but if you want I'll rewrite it.

Reply:
I would, because I can't understand the code written at a glance atm. But perhaps @joachimvh will be more lenient 🙂

Reply:
No, you're right, the difference in readability is high, so I changed it.